akka-javasdk/src/main/java/akka/javasdk/workflow/Workflow.java

```diff
@@ -193,9 +218,17 @@ interface Builder<S> {
 
       /**
        * Pause the workflow execution and wait for an external input, e.g. via command handler.
+       * After a some time the workflow will passivate and clean up resources.
        */
       TransitionalEffect<Void> pause();
 
+      /**
+       * Pause the workflow execution and wait for an external input, e.g. via command handler.
+       * @param keepAlive If true, the workflow will not passivate and will keep its resources alive.
+       *                  Important when workflow has a long-running background process.
+       */
+      TransitionalEffect<Void> pause(boolean keepAlive);
```

Comment:
It might be interesting to have a keepAlive anyway, also for cases where we explicitly, for some reason, don't want to passivate. 

But if the reason is to support background processes, then we can simply check if there is one running instead of using a boolean. 

On the other hand, we can also say that if you pause a workflow, you also pause the background process. Unless you explicitly call this  method.

Reply:
> then we can simply check if there is one running instead of using a boolean

yes, but then the background process should have a method `isRunning` or sth.

> On the other hand, we can also say that if you pause a workflow, you also pause the background process. Unless you explicitly call this method.

using `pause` is also forced by the compiler, it's the only "transition" that makes sense. So for not pausing we would need another transition type.